File: DemoMethods/src/main/java/net/ravendb/demo/indexes/TransformerLastFm.java
package net.ravendb.demo.indexes;

import net.ravendb.client.indexes.AbstractTransformerCreationTask;

public class TransformerLastFm extends AbstractTransformerCreationTask {

    public TransformerLastFm() {
        transformResults = "from result in results " +
                "select new { " +
                "  Artist = result.Artist, " +
                "  TimeStamp = result.TimeStamp, " +
                "  Tags = result.Tags, " +
                "  TrackId = result.TrackId, " +
                "  Title = result.Title " +
                "}";
    }

}

File: DemoServer/Indexes/TransformerLastFm.cs
using System.Linq;
using DemoServer.Entities;
using Raven.Client.Documents.Transformers;

namespace DemoServer.Indexes
{
    public class TransformerLastFm : AbstractTransformerCreationTask<LastFm>
    {
        public TransformerLastFm()
        {
            TransformResults = results => from result in results
                                          select new
                                          {
                                              result.Artist,
                                              result.TimeStamp,
                                              result.Tags,
                                              result.TrackId,
                                              result.Title
                                          };
        }
    }

}